tsl/src/compression/compress_utils.c

```diff
@@ -507,6 +514,14 @@ tsl_compress_chunk(PG_FUNCTION_ARGS)
 		if (!compress_remote_chunk(fcinfo, chunk, if_not_compressed))
 			PG_RETURN_NULL();
 
+		/*
+		 * Updating the chunk compression status of the Access Node AFTER executing remote
+		 * compression. In the event of failure, the compressed status will NOT be set. The
+		 * distributed compression policy will attempt to compress again, which is idempotent, thus
+		 * the metadata are eventually consistent.
+		 */
```

Comment:
Do you need to call `compress_remote_chunk` if compress flag is set? If you have a failure in compressing all chunks, you will get an error and not set the flag, so if the flag is set, it should be a guarantee that the access node has seen and processed all responses.

Reply:
It's always possible that a Data Node chunk has been manually decompressed. The remote `compress_chunk()` function should guarantee that the replica chunks are actually compressed when it returns.